### ISSUE TITLE
Signin with Google/First sync with gmail is buggy

### ISSUE THREAD

**Issue by gvkhna** [twentyhq/twenty#6940]
## Bug Description

There's reproducible behavior and issues with the following steps.

1. Signin with Google
2. Oauth gmail, server starts syncing...
3. Go into email settings, turn off exclude @gmail, turn off exclude group emails
4. Wait for finish sync, most emails are missing.
5. Turn on/off these settings doesn't do anything to try to trigger a fresh sync.

Solution:
6. Remove google account -> Connect account again
7. Turn off exclude @gmail again

Then it seems to be working fine, but that initial disconnect/connect seems to be required. I read of other's encountering it as well and I reset local db multiple times, the first sync just doesn't seem to work correctly and respect the settings having to do with excluding gmail etc.

**Comment by FelixMalfait** [comment]
@bosiraphael could you please look into this? thanks!

**Comment by FelixMalfait** [comment]
As discussed with @Bonapara this is probably a UX issue. We should maybe push users to disconnect/reconnect their inbox after they change a setting if they want that setting to be applied retroactively and not just on future emails

**Comment by gvkhna** [comment]
It’s kind of tricky because by default the sync starts with that at default. So it’s possible to miss something if there’s no way to force a resync. I think there should be functionality to force a fresh resync as if you had added and removed. Otherwise it should ask before starting the sync. Many companies do have customers that have gmail accounts they would want in CRM.

**Comment by FelixMalfait** [comment]
yes I agree @gvkhna, you would probably like that issue which partially answers your suggestion https://github.com/twentyhq/twenty/issues/5912
